packages/docusaurus/src/client/routeContext.tsx

```diff
@@ -30,6 +30,7 @@ function mergeContexts({
     return value;
   }
 
+  // TODO deep merge this
```

Comment:
not sure we really want to deep merge

In case there are multiple layers I'd rather ensure that laters do not override each others, there's no good use-case to do so that I can think of
We can keep the comment for now and figure this out later anyway

Reply:
I don't think it would be immediately useful for content plugins, but maybe in userland? Deep merging sounds more natural about how these stores should behave